lib/matplotlib/tests/test_axes.py

```diff
@@ -1706,63 +1706,120 @@ def test_hist2d_transpose():
     ax.hist2d(x, y, bins=10, rasterized=True)
 
 
-@image_comparison(baseline_images=['scatter', 'scatter'])
-def test_scatter_plot():
-    fig, ax = plt.subplots()
-    data = {"x": [3, 4, 2, 6], "y": [2, 5, 2, 3], "c": ['r', 'y', 'b', 'lime'],
-            "s": [24, 15, 19, 29]}
-
-    ax.scatter(data["x"], data["y"], c=data["c"], s=data["s"])
-
-    # Reuse testcase from above for a labeled data test
-    fig, ax = plt.subplots()
-    ax.scatter("x", "y", c="c", s="s", data=data)
+class TestScatter(object):
```

Comment:
Is there a particular reason, you put the tests into a class?

Reply:
I am no expert in test writing but at least it seems clearer to me to group all the tests related to the same “main method/function/stuff” into a single place (a very strong argument, huh \o/). Besides if I am correct, you can then do things like
```
pytest lib/matplotlib/tests/test_axes.py::TestScatter
``` 
to run only the relevant subset of the tests (rather than having to find all of them with autocompletion), which can be significantly faster (and more pleasant when you are still in the debugging phase) than running the whole test file (e.g. 1.5 s vs. ~80 s on my current local machine).

But TBH, I could not find any guideline for Matplotlib that suggests to use classes rather than dumping all test functions at the root of the test file : some test files like [test_ticker.py](https://github.com/matplotlib/matplotlib/blob/fb98a9a9a175d4912d88377ffd80bab24cff5fc1/lib/matplotlib/tests/test_ticker.py) use them a lot while other files do not (like [test_axes.py](https://github.com/matplotlib/matplotlib/blob/fb98a9a9a175d4912d88377ffd80bab24cff5fc1/lib/matplotlib/tests/test_axes.py) ^^).

Reply:
I don‘t think there is a policy on using or nor using Test classes. So you‘re probably right whatever you do.

Personally, I would not use classes (flat is better than nested). The grouping is already done by name ˋtest_scatter...ˋ. You can use keyword expressions ˋpytest -k scatterˋ to select a subset of tests.

Reply:
Wha????  That’s amazing.

Reply:
I prefer test functions over test classes (unless you need to scope fixtures to the class).

Reply:
But that is not worth blocking the PR over.